pennylane/labs/tests/resource_estimation/test_resource_container.py

```diff
@@ -0,0 +1,315 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Test base Resource class and its associated methods
+"""
+# pylint:disable=protected-access, no-self-use
+import copy
+from collections import defaultdict
+
+import pytest
+
+import pennylane as qml
+from pennylane.labs.resource_estimation.resource_container import (
+    CompressedResourceOp,
+    Resources,
+    _combine_dict,
+    _scale_dict,
+    add_in_parallel,
+    add_in_series,
+    mul_in_parallel,
+    mul_in_series,
+)
+
+
+class TestCompressedResourceOp:
```

Comment:
These tests will need to be rewritten once our other PRs are merged. We decided to build `CompressedResourceOp` from `ResourceOperator` instead of PennyLane operators, and the tests will need to be updated. Another issue is that we haven't added a resource operator for QSVT, so we should rewrite these tests to only use operators we have access to.

I'm fine with the tests as they are for this PR. Once the `resource_constructor` branch is merged these tests will fail since there will be a check to make sure a resource operator is passed into the init of the compressed resource op. Maybe we can fix these tests within that PR?

Reply:
I think this is a good idea. Lets merge the constructor PR into this one, then I can modify the test and merge upwards 👍🏼